analyzer/src/test/scala/com/avsystem/commons/analyzer/ThrownExceptionNotInFunctionTest.scala

```diff
@@ -0,0 +1,209 @@
+  settings.pluginOptions.value ++= List("AVSystemAnalyzer:-discardedMonixTask")
+
+
+  test("Testing simple methods") {
+    assertErrors(1,
+      //language=Scala
+      """
+        |object whatever {
+        |  def ex: Exception = ???
+        |
+        |  def f0(x1: Int => Int) = ???
+        |
+        |  //not ok
+        |  f0(throw ex)
+        |
+        |  //ok
+        |  f0(identity)
+        |}""".stripMargin)
+  }
+
+  Seq(
+    ("Option[_]", "map"),
+    ("Option[_]", "flatMap"),
+    ("List[_]", "map"),
+    ("Seq[_]", "map"),
+    ("Set[_]", "map"),
+    ("Map[_, _]", "map"),
+    ("scala.concurrent.Future[_]", "map"),
+    ("scala.concurrent.Future[_]", "flatMap"),
+    ("scala.util.Try[_]", "map"),
+    ("Either[_, _]", "map"),
+    ("monix.eval.Task[_]", "map"),
+    ("monix.eval.Task[_]", "flatMap"),
+    ("com.avsystem.commons.misc.Opt[_]", "map"),
+    ("String => Int", "andThen"),
+    ("String => Nothing", "andThen"),
+    ("Nothing => Nothing", "andThen"),
+    ("String => Int", "compose"),
+    ("Seq[_]", "foreach"),
+  ).foreach { case (tpe, function) =>
+    test(s"Testing method $function of $tpe") {
+      assertErrors(10,
+        //language=Scala
+        s"""
+           |object whatever {
+           |  implicit val ec: scala.concurrent.ExecutionContext = ??? // for Future
+           |
+           |  def sth: $tpe = ???
+           |  def ex: Exception = ???
+           |
+           |  // errors from these
+           |  sth.$function(throw ex)
+           |
+           |  {
+           |    println(""); sth.$function(throw ex)
+           |  }
+           |
+           |  if (true) sth.$function(throw ex) else sth.$function(throw ex)
+           |
+           |  try sth.$function(throw ex) catch {
+           |    case _: Exception => sth.$function(throw ex)
+           |  } finally sth.$function(throw ex)
+           |
+           |  Seq(1, 2, 3).foreach(_ => sth.$function(throw ex))
+           |
+           |  while (true) sth.$function(throw ex)
+           |
+           |  do sth.$function(throw ex) while (true)
+           |
+           |  // no errors from these
+           |  sth.$function(_ => throw ex)
+           |
+           |  {
+           |    println(""); sth.$function(_ => throw ex)
+           |  }
+           |
+           |  if (true) sth.$function(_ => throw ex) else sth.$function(_ => throw ex)
+           |
+           |  try sth.$function(_ => throw ex) catch {
+           |    case _: Exception => sth.$function(_ => throw ex)
+           |  } finally sth.$function(_ => throw ex)
+           |
+           |  Seq(1, 2, 3).foreach(_ => sth.$function(_ => throw ex))
+           |
+           |  while (true) sth.$function(_ => throw ex)
+           |
+           |  do sth.$function(_ => throw ex) while (true)
+           |}
+           |
+           |""".stripMargin
+      )
+    }
+  }
+
+  test("Testing multiple arguments") {
+    assertErrors(26,
```

Comment:
Debugging why an `assertErrors(26, ...` failed is impossible without editing the tests. Let's work on the developer experience in this test by using sth like `org.scalatest.Inspectors` in conjunction with `assertErrors(1` and `assertNoErrors` (example, I'm sure you can suggest the best solution available).